src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -51,10 +51,11 @@ bool syncMode = false;
 namespace impl {
 
 SolverInterfaceImpl::SolverInterfaceImpl(
-    std::string participantName,
-    int         accessorProcessRank,
-    int         accessorCommunicatorSize,
-    void *      communicator)
+    std::string        participantName,
+    const std::string &configurationFileName,
```

Comment:
Why do we treat `participantName` and `configurationFileName` differently? i.e. `std::string` vs. `const std::string &`?

Reply:
We need to store the `participantName` inside the object, so we use call-by-value and move construct the member which holds it in the constructor of the `SolverInterfaceImpl`.

The configuration file name does not need to be stored as we simply call `configure` in the constructor of `SolverInterfaceImpl`. So we simply pass it by const ref.

This is not important and I wouldn't mind changing both to pass by value for the sake of readability.

Reply:
Thanks for the explanation. I merely asked for learning, but you might have a point. A user not deep into C++ might wonder if she needs to treat `participantName` and `configurationFileName` differently when calling the constructor.

Reply:
The user does not need to treat both differently.
We just do things slightly different in the implementation.